### PR TITLE
Extracts Notice logic into the JetpackDisconnectedNotice component

### LINKED ISSUE #64546: Marketplace: Implement NoticesSection Component
### Details
Lets encapsulate Notices, Announcements and Upgrade Nudge functionality into a `NoticeSection` component.
Screen Shot 2022-06-10 at 6 14 02 PM

pdh6GB-14m-p2

### Checklist

- [ ] Should encapsulate state selection for notices ~and the upgrade nudge~.
- [ ] Should be able to render the "Site cannot be accessed" notice.
- [ ] ~Should render the upgrade nudge.~

### Related

_No response_

### PR DESCRIPTION
**Pull request by Rolilink**
#### Proposed Changes

* Extracts the state management logic for the site disconnect section.
* Adds it into `JetpackDisconnectedNotice `.

#### Testing Instructions

* Apply D86035-code.
* Navigate to the plugins marketplace.
* The disconnected site notice should appear.

#### Pre-merge Checklist

- [ ] ~[Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?~
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~
- [ ] ~Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?~

Fixes #64546

### CODE REVIEW

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 1]
```diff
@@ -305,25 +291,7 @@ const PluginsBrowser = ( { trackPageViews = true, category, search, hideHeader }
...
-					<NoticeAction
-						onClick={ trackSiteDisconnect }
-						href={ `/settings/disconnect-site/${ selectedSite.slug }?type=down` }
-					>
-						{ translate( 'I’d like to fix this now' ) }
-					</NoticeAction>
-				</Notice>
-			) }
-
+			<PluginsNoticeSection />
```
Comment (lsl) [inline client/my-sites/plugins/plugins-browser/index.jsx]: Could we rename this to something like `JetpackDisconnectedNotice`? Or was the intent other notices go here too (like the paid nudge ones etc?)

**Review by lsl — APPROVED** [review]
LGTM

Screenshot 2022-08-17 at 14-23-15 Plugins ‹ test — WordPress com

**Review by paulopmt1 — APPROVED** [review]
LGTM! It's always good to see refactors like this @Rolilink!

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 1 above)
Comment (Rolilink) [reply client/my-sites/plugins/plugins-browser/index.jsx]: Was thinking the same last night @lsl, will do it before merging

**Pull request merged**